tokio-net/src/driver/reactor.rs

```diff
@@ -349,17 +346,15 @@ impl fmt::Debug for Reactor {
 // ===== impl Handle =====
 
 impl Handle {
-    #[doc(hidden)]
-    #[deprecated(note = "semantics were sometimes surprising, use Handle::default()")]
-    pub fn current() -> Handle {
-        // TODO: Should this panic on error?
-        HandlePriv::try_current()
-            .map(|handle| Handle {
-                inner: Some(handle),
-            })
-            .unwrap_or(Handle {
-                inner: Some(HandlePriv { inner: Weak::new() }),
-            })
+    /// Get a handle to the current reactor.
+    ///
+    /// # Returns
+    ///
+    /// - `Ok<Handle>` if there is reactor set for the current thread
+    /// - `Err` if there is no reactor set for the current thread
+    pub fn current() -> io::Result<Handle> {
+        let inner = HandlePriv::try_current()?;
+        Ok(Handle { inner: Some(inner) })
```

Comment:
do we need `Option` in inner now that it is no longer lazily initialized?

Reply:
Oh probably not. I should be able to remove the `Option`.

Reply:
@DoumanAsh Thanks for pointing this out. This led to some further changes and a lot of additional code was able to be removed.